diff -Naur homer-api/scripts/homer_mysql_rotate.pl homer-api-aft/scripts/homer_mysql_rotate.pl
--- homer-api/scripts/homer_mysql_rotate.pl	2016-09-14 14:20:10.000000000 +0300
+++ homer-api-aft/scripts/homer_mysql_rotate.pl	2016-10-22 17:52:51.345174753 +0300
@@ -150,7 +150,7 @@
         my $sth = $db->prepare($query);
         $sth->execute();
         my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = gmtime(time() - 86400*$rotation_horizon);
-        my $oldest = sprintf("%04d%02d%02d",($year+=1900),(++$mon),$mday,$hour);
+        my $oldest = sprintf("%04d%02d%02d%02d",($year+=1900),(++$mon),$mday,$hour);
         $oldest+=0;
         while(my @ref = $sth->fetchrow_array()) {
            my $table_name = $ref[0];
