diff -Naur sngrep-1.4.0/configure.ac sngrep-1.4.0-aft/configure.ac
--- sngrep-1.4.0/configure.ac	2016-08-10 14:39:45.000000000 +0300
+++ sngrep-1.4.0-aft/configure.ac	2016-10-18 01:04:50.284184492 +0300
@@ -61,6 +61,10 @@
 	    AC_MSG_ERROR([ You need to have libncursesw installed to compile sngrep.])
 	])
 
+	AC_CHECK_LIB([tinfo], [cbreak], [], [
+	    AC_MSG_ERROR([ You need to have tinfo installed to compile sngrep.])
+	])
+
 	AC_CHECK_LIB([panelw], [new_panel], [], [
 	    AC_MSG_ERROR([ You need to have ncurses panelw library installed to compile sngrep.])
 	])
